components/org.wso2.carbon.identity.media.core/src/main/java/org/wso2/carbon/identity/media/core/StorageSystemManager.java

```diff
@@ -274,17 +281,109 @@ public void validateFileUploadMediaTypes(String mediaTypePathParam, String conte
+                }
+                return userIdFromUserName;
+            }
+        } catch (UserStoreException e) {
+            throw new StorageSystemServerException("Error occurred while retrieving the userstore manager to resolve" +
+                    " id for the user: " + username, e);
+        }
+        return null;
+    }
+
+    /**
+     * Validates if the file size of the media to be uploaded doesn't exceed the maximum allowed file size.
+     *
+     * @param inputStream The media as an input.
+     * @throws StorageSystemException Exception related to validating media size.
+     */
+    public void validateMediaSize(InputStream inputStream) throws StorageSystemException {
 
+        String envForMediaSize = System.getenv(CONFIGURABLE_MAXIMUM_MEDIA_SIZE_IN_BYTES);
+        int allowedMaximumMediaSize;
+        if (StringUtils.isNotBlank(envForMediaSize)) {
+            allowedMaximumMediaSize = Integer.parseInt(envForMediaSize);
+        } else {
+            allowedMaximumMediaSize = StorageSystemUtil.getMediaMaximumSize();
+        }
+
+        try {
+            byte[] mediaByteArray = IOUtils.toByteArray(inputStream);
+            if (mediaByteArray.length > allowedMaximumMediaSize) {
+                throw new StorageSystemClientException(String.format("The uploaded media size: %skb exceeds the " +
+                                "maximum allowed file size: %skb", mediaByteArray.length / 1000.0,
+                        allowedMaximumMediaSize / 1000.0));
+            }
+            inputStream.reset();
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("The file size: %skb of the media to be uploaded doesn't exceed the " +
+                                "maximum allowed file size: %skb. Hence proceeding with the upload.",
+                        mediaByteArray.length / 1000.0, allowedMaximumMediaSize / 1000.0));
+            }
+        } catch (IOException e) {
+            throw new StorageSystemServerException("Error occurred while calculating media size.", e);
+        }
+    }
+
+    private String getMediaStoreType() {
+
+        String envForMediaStoreType = System.getenv(CONFIGURABLE_MEDIA_STORE_TYPE);
+        if (StringUtils.isNotBlank(envForMediaStoreType)) {
+            return envForMediaStoreType;
+        } else {
+            return StorageSystemUtil.getMediaStoreType();
+        }
+    }
+
+    private StorageSystem getStorageSystem(String storageType) throws StorageSystemServerException {
+
+        StorageSystemFactory storageSystemFactory =
+                MediaServiceDataHolder.getInstance().getStorageSystemFactories().get(storageType);
+        if (storageSystemFactory == null) {
+            throw new StorageSystemServerException(String.format("Unable to obtain StorageSystemFactory for " +
+                    "configured media store type: %s", storageType));
+        }
+        return storageSystemFactory.getInstance();
```

Comment:
here also since storageSystemFactory is an instance variable, its not correct to call getInstance() on that. can we rename the getInstace method to getStorageSystem or something meaningful?

Reply:
fixed